File: 1582. special-positions-in-a-binary-matrix.java
class Solution {
    public int numSpecial(int[][] mat) {
        int m = mat.length, n = mat[0].length;
        int[] row = new int[m];
        int[] col = new int[n];
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                if (mat[i][j] == 1) {
                    row[i]++;
                    col[j]++;
                }
            }
        }

        int ans = 0;
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                if (mat[i][j] == 1 && row[i] == 1 && col[j] == 1) {
                    ans++;
                }
            }
        }

        return ans;
    }
    
    public static void main(String[] args) {
        Solution sol = new Solution();
        
        // Example 1:
        int[][] mat1 = {{1,0,0},{0,0,1},{1,0,0}};
        int output1 = sol.numSpecial(mat1);
        System.out.println("Example 1: Input: mat = [[1,0,0],[0,0,1],[1,0,0]]");
        System.out.println("Output: " + output1);
        System.out.println("Expected: 1");
        System.out.println();
        
        // Example 2:
        int[][] mat2 = {{1,0,0},{0,1,0},{0,0,1}};
        int output2 = sol.numSpecial(mat2);
        System.out.println("Example 2: Input: mat = [[1,0,0],[0,1,0],[0,0,1]]");
        System.out.println("Output: " + output2);
        System.out.println("Expected: 3");
    }
}

File: 1582. special-positions-in-a-binary-matrix.py
class Solution:
    def numSpecial(self, mat):
        m, n = len(mat), len(mat[0])
        row = [0] * m
        col = [0] * n

        for i in range(m):
            for j in range(n):
                if mat[i][j] == 1:
                    row[i] += 1
                    col[j] += 1

        return sum(
            mat[i][j] == 1 and row[i] == 1 and col[j] == 1
            for i in range(m)
            for j in range(n)
        )

if __name__ == "__main__":
    sol = Solution()
    
    # Example 1:
    mat = [[1,0,0],[0,0,1],[1,0,0]]
    output = sol.numSpecial(mat)
    print(f"Example 1: Input: mat = {mat}")
    print(f"Output: {output}")
    print(f"Expected: 1")
    print()
    
    # Example 2:
    mat = [[1,0,0],[0,1,0],[0,0,1]]
    output = sol.numSpecial(mat)
    print(f"Example 2: Input: mat = {mat}")
    print(f"Output: {output}")
    print(f"Expected: 3")